Add download dashboard option

OriBenHur-Any opened the pull request.

OriBenHur-Any commented:
@igaldahan can you please approve my PR

ghost commented:
@sashas-anyvision can you please approve my changes

sashas-anyvision commented on compose-oneliner.sh:
```diff
@@ -12,9 +12,9 @@ HOME_DIR=`eval echo ~$(logname)`
 DOCKER_COMPOSE_DIR=${HOME_DIR}/docker-compose
 COMPOSE_BASH_URL="https://github.com/AnyVisionltd"
 BRANCH=$1
-PRODUCT=$2
```
please work on commit message .. 
how does changing environment variable helps ? can you please describe the problem and how does the solution solves it ?

OriBenHur-Any replied:
I have optional commits sso they need to come last.

sashas-anyvision commented on a-v.sh:
```diff
@@ -62,7 +67,7 @@ git clone --recurse-submodules  https://github.com/AnyVisionltd/compose-oneliner
 pushd /opt/compose-oneliner && chmod u+x /opt/compose-oneliner/compose-oneliner.sh
 
-exec ./compose-oneliner.sh ${BRANCH} ${PRODUCT} ${GIT} ${TOKEN}
+exec ./compose-oneliner.sh ${BRANCH} ${TOKEN} ${GIT} ${PRODUCT}
```
this shoiuld be squashed with previous commit

sashas-anyvision commented on README.md:
```diff
@@ -1,4 +1,4 @@
 # compose-oneliner
-wget -qO- https://raw.githubusercontent.com/AnyVisionltd/compose-oneliner/development/a-v.sh | bash -s -- -b \<BRANCH\> [-p \<PRODUCT\>] [-g \<GIT\>]
+wget -qO- https://raw.githubusercontent.com/AnyVisionltd/compose-oneliner/development/a-v.sh | bash -s -- -b \<BRANCH\> -t \<TOKEN\>[-p \<PRODUCT\>] [-g \<GIT\>]
```
this should be squashed with first commit

sashas-anyvision commented on README.md:
```diff
@@ -1,4 +1,5 @@
-wget -qO- https://raw.githubusercontent.com/AnyVisionltd/compose-oneliner/development/a-v.sh | bash -s -- -b \<BRANCH\> -t \<TOKEN\>[-p \<PRODUCT\>] [-g \<GIT\>]
 
-    * the default git is https://github.com/AnyVisionltd
+wget -qO- <https://raw.githubusercontent.com/AnyVisionltd/compose-oneliner/development/a-v.sh> | bash -s -- -b \<BRANCH\> -t \<TOKENP\> [-p \<PRODUCT\>] [-g \<GIT\>]
```
also this

sashas-anyvision commented on a-v.sh:
```diff
@@ -67,7 +67,7 @@ git clone --recurse-submodules  https://github.com/AnyVisionltd/compose-oneliner
 pushd /opt/compose-oneliner && chmod u+x /opt/compose-oneliner/compose-oneliner.sh
 
-exec ./compose-oneliner.sh ${BRANCH} ${TOKEN} ${GIT} ${PRODUCT}
+exec ./compose-oneliner.sh ${BRANCH} ${TOKEN} ${PRODUCT} ${GIT} 
```
also this

sashas-anyvision commented on compose-oneliner.sh:
```diff
@@ -13,8 +13,8 @@ DOCKER_COMPOSE_DIR=${HOME_DIR}/docker-compose
-COMPOSE_REPO_GIT="${3:-docker-compose}.git"
-PRODUCT=$4
+PRODUCT=$3
+COMPOSE_REPO_GIT="${4:-docker-compose}.git"
```
why ?

sashas-anyvision commented on compose-oneliner.sh:
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env bash
+set -x
```
why do we need this ?

OriBenHur-Any replied:
we don't it was left out on the last commit.
i've committed all the above for my on testing the last commit is the one with all the changes.

sashas-anyvision reviewed (changes requested):
i did not review all commits .. please rearrange commits and resquash the code 
in every commit please specify the following:
 please describe the problem and how does the solution solves it ?

sashas-anyvision commented:
I did not see any changes .. please requash commits
[image: Employee profile]
Sasha Solganik
Backend Team Leader
Development Team
M +972 543348368 <+972%20543348368>
E alexanders@anyvision.co
W www.anyvision.co


On Sun, Oct 6, 2019 at 2:08 PM OriBenHur-Any <notifications@github.com>
wrote:

> *@OriBenHur-Any* commented on this pull request.
> ------------------------------
>
> In compose-oneliner.sh
> <https://github.com/AnyVisionltd/compose-oneliner/pull/8#discussion_r331783205>
> :
>
> > @@ -1,4 +1,5 @@
>  #!/usr/bin/env bash
> +set -x
>
> we don't it was left out on the last commit.
> i've committed all the above for my on testing the last commit is the one
> with all the changes.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/AnyVisionltd/compose-oneliner/pull/8?email_source=notifications&email_token=AL2OXVMLKU3FQQL37YXHQA3QNHBJRA5CNFSM4I5ELDC2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCHAI47A#discussion_r331783205>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AL2OXVP4A7BCXUTX72SJFM3QNHBJRANCNFSM4I5ELDCQ>
> .
>

ghost commented:
I did just now, all the commits are now squashed together

sashas-anyvision commented on a-v.sh:
```diff
@@ -40,6 +65,11 @@ if [ -z $PRODUCT ]; then
     PRODUCT="BT"
 fi
 
+# if [ -z $GIT ]; then
```
what is this ?

sashas-anyvision commented on compose-oneliner.sh:
```diff
@@ -49,17 +50,24 @@ apt -qq update
 
 echo 'xhost +SI:localuser:root' | tee /etc/profile.d/xhost.sh
 usermod -aG docker $(logname)
-chown -R $(logname):$(logname) ${DOCKER_COMPOSE_DIR}
```
why is this removed ?

sashas-anyvision commented on compose-oneliner.sh:
```diff
@@ -49,17 +50,24 @@ apt -qq update
-chown -R $(logname):$(logname) ${DOCKER_COMPOSE_DIR}
+
 pushd ${DOCKER_COMPOSE_DIR}/${BRANCH}
+ver=`grep -F /api: ${DOCKER_COMPOSE_DIR}/${BRANCH}/docker-compose.yml | grep -Po 'api:\K[^"]+'`
```
what problem does it solve ?

sashas-anyvision reviewed (changes requested):
please come to me ..  we will do it once together.

Commit:
installer: fix parsing gcr token as git repo parameter

Previously git parameter was specified before gcr token parameter. This
resulted in a behaviour that if we did not specify GCR token and just
specified git repo, git repo name was considered as  GCR tokken by the
script, thus token was incorrect, and git was not specified.
Solution is to put all optional parameters at the end.

Commit:
installer: add debug options for script.

Allow to debug script by provoding --debug, also decrease ansible debug
a bit.

Commit:
installer: add ability to download dashboard during installation.

Now we can also download dashboard via providing --download-dashboard
parameter to installer script.

Commit:
installer: add message to reboot system

System must be rebooted after installation due to the fact that we
are installing nvidia drivers during installation, hence we need to
reboot the system

Commit:
installer: provide help and readme

Commit:
installer: make installer script executable

Commit:
remove invisible character in help function
